packages/firestore/lib/src/utils/push_id_generator.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:math';
+
+/// Utility class for generating Firebase child node keys.
+///
+/// Since the Flutter plugin API is asynchronous, there's no way for us
+/// to use the native SDK to generate the node key synchronously and we
+/// have to do it ourselves if we want to be able to reference the
+/// newly-created node synchronously.
+///
+/// This code is based on a Firebase blog post and ported to Dart.
+/// https://firebase.googleblog.com/2015/02/the-2120-ways-to-ensure-unique_68.html
+class PushIdGenerator {
+  static const String PUSH_CHARS =
```

Comment:
Can this be private?

Reply:
This is not a public API, it is only used within the package.

Reply:
I could make it an underscore class and use `part` if you think that's better...

Reply:
Oh, you're right. For some reason I though this was brought into the global name space with part of. But you're right, it is not.